test-case/check-audio-equalizer.sh

```diff
@@ -52,95 +52,95 @@ func_test_eq()
         dloge "Equalizer setting failure with $conf"
         return 1
     fi
-    # test with aplay
-    dlogc "aplay -D$dev -f $fmt -c $channel -r $rate /dev/zero -d $duration"
-    aplay -D$dev -f $fmt -c $channel /dev/zero -d $duration
+
+    dlogc "$cmd -D $dev -f $fmt -c $channel -r $rate -d $duration $dummy_file"
+    $cmd -D $dev -f $fmt -c $channel -r $rate -d $duration $dummy_file
     if [ $? -ne 0 ]; then
         dloge "Equalizer test failure with $conf"
         return 1
     fi
     sleep 1
 }
 
+# this function performs IIR/FIR filter test
+# param1 must be must be iir or fir
+func_test_filter()
+{
+    local testfilter=$1
+    dlogi "1. Get amixer control id for $testfilter"
+    Filterid=`amixer -D hw:$sofcard controls | grep -i eq${testfilter}| head -1| sed 's/numid=\([0-9]*\),.*/\1/'`
```

Comment:
Backticks are deprecated https://mywiki.wooledge.org/BashFAQ/082

This was reported by shellcheck, please look at the Travis output. I'll review the rest after you have.

Reply:
BTW `sed` has a built-in `grep` feature:
```shell
amixer -D default controls | sed -n -e "/eq${testfilter}/I "'s/numid=\([0-9]*\),.*/\1/p'
```